static/js/message_events.js

```diff
@@ -149,6 +149,19 @@ exports.update_messages = function update_messages(events) {
             const compose_stream_name = compose_state.stream_name();
             const orig_topic = util.get_edit_event_orig_topic(event);
 
+            if (page_params.realm_allow_edit_history) {
+                const topic_edit_history = {
+                    timestamp: event.edit_timestamp,
+                    user_id: event.user_id,
+                    prev_topic: orig_topic,
+                };
+
+                if (msg.edit_history === undefined) {
+                    msg.edit_history = [];
+                }
+                msg.edit_history = [topic_edit_history].concat(msg.edit_history);
+            }
```

Comment:
Looking at this more, I think we should (1) extract this block to a common `update_edit_history(msg, event)` that can be called once, outside the conditionals for topic or message editing, so we don't get 2 edit history events if the topic and content were edited at once (or soon, if the stream was edited too).

Does that make sense to you?

Reply:
My bad, I thought your comment was referring to the tests. 

I made this change, but should I let the tests stay as it is? The test first edits a topic and then the content, though. Should I change that to make both edits simultaneously?

Reply:
I think this is fine; but ideally we'd test both simultaneous and separate.